### PR TITLE
Rename is_system_auditor to is_platform_auditor

### PR DESCRIPTION
**Pull request by relrod**
Signed-off-by: Rick Elrod <rick@elrod.me>

### CODE REVIEW

**Review by john-westcott-iv — COMMENTED** [review]
My only concern with this is the authenticators. i.e. if I'm an awx customer and I already have the is_system_auditor flag defined in my saml/oidc will this cause the customer to have to change something in their provider for the functionality to work. Keycloak is a net new authenticator so I am fine with that one changing.

**Comment by AlanCoding** [comment]
Can we turn it into a setting? Then moving forward @john-westcott-iv, can AWX continue to respect the current field, and apps can change for their own expectations.

File: ansible_base/lib/dynamic_config/dynamic_settings.py [hunk 1]
```diff
@@ -139,6 +139,8 @@
     SOCIAL_AUTH_STRATEGY = "ansible_base.authentication.social_auth.AuthenticatorStrategy"
     SOCIAL_AUTH_LOGIN_REDIRECT_URL = "/"
 
+    ANSIBLE_BASE_SOCIAL_AUDITOR_FLAG = "is_system_auditor"
```
Comment (john-westcott-iv) [inline ansible_base/lib/dynamic_config/dynamic_settings.py:142]: Should we change this to default to `is_platform_auditor` as a "going forward" approach?
Also, we should add this to the docs for authentication to let people know they need to set it if required?

Comment (AlanCoding) [reply ansible_base/lib/dynamic_config/dynamic_settings.py:142]: My logic here is that "is_system_auditor" isn't going away, it just gives an indicator of "local" permissions, not permissions that are shared with some other system. And maybe we don't migrate it at all, and move towards role mappings instead. So I still like this as the default. But maybe it should be added in docs.

**Review by john-westcott-iv — APPROVED** [review]

File: ansible_base/authentication/authenticator_plugins/oidc.py [hunk 2]
```diff
@@ -207,7 +207,9 @@ class AuthenticatorPlugin(SocialAuthMixin, OpenIdConnectAuth, AbstractAuthentica
     configuration_encrypted_fields = ['SECRET']
 
     def extra_data(self, user, backend, response, *args, **kwargs):
-        for perm in ["is_superuser", "is_system_auditor"]:
+        from ansible_base.lib.utils.settings import get_setting
```
Comment (relrod) [inline ansible_base/authentication/authenticator_plugins/oidc.py]: Does this have to be a local import or can it move up?

Comment (AlanCoding) [reply ansible_base/authentication/authenticator_plugins/oidc.py]: This is in plugins, which is why I'm hesitant to do it top-level.

Comment (AlanCoding) [reply ansible_base/authentication/authenticator_plugins/oidc.py]: ok, well I'll give it a shot, and I assume if it still passes checks that's ok.

**Pull request merged**